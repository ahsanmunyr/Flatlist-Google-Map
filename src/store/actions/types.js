export const VENUE_DATA = "VENUE_DATA";
export const VENUE_DATA_MORE = "VENUE_DATA_MORE";
export const VENUE_ERROR = "VENUE_ERROR";
export const VENUE_DATA_CLEAR = "VENUE_DATA_CLEAR";
