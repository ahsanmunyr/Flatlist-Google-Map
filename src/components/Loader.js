import { StyleSheet, Text, View ,ActivityIndicator} from 'react-native'
import React from 'react'
import { responsiveFontSize } from 'react-native-responsive-dimensions'
import colors from '../assets/colors/colors'

export default function Loader({color,size}) {
  return (
    <View style={styles.con}>
      <ActivityIndicator size={responsiveFontSize(size?size:4)} color={color ? color :colors.themeblue}/>
    </View>
  )
}

const styles = StyleSheet.create({
    con:{
        flex:1,
        justifyContent:'center',
        alignItems:'center',
    }
})