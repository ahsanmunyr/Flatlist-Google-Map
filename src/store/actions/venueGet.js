import axios from 'axios';
import * as types from './types';
import config from '../../config/config';

export const venueGet = (page, loadMore) => async dispatch => {
  try {
    const response = await axios.get(`${config.base_url}?p=${page}`);
    console.log('API CALL');
    if (response?.data?.status) {
      if (loadMore == 'loadmore') {
        dispatch({
          type: types.VENUE_DATA_MORE,
          payload: response.data,
        });
      } else {
        dispatch({
          type: types.VENUE_DATA,
          payload: response.data,
        });
      }
    }
  } catch (e) {
    dispatch({
      type: types.VENUE_ERROR,
      payload: 'ERROR',
    });
    console.log(e.response.data);
  }
};
