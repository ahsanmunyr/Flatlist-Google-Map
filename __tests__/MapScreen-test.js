import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../src/components/Loader';
import { venueGet } from '../src/store/actions/venueGet';
test('renders loader component currectly', () => {
  const loaderView = renderer.create(<Loader />).toJSON();
  console.log(loaderView.props,  "loaderView");
  expect(loaderView).toMatchSnapshot();
});
