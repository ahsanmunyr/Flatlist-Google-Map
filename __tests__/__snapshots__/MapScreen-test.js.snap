// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader component currectly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#8b7f79"
    size={61.191865835619396}
  />
</View>
`;
